packages/meditrak-server/src/routes/exportSurveyResponses.js

```diff
@@ -227,66 +227,64 @@ export async function exportSurveyResponses(req, res) {
+        ]);
+      }
 
-        // Add any questions that are in survey responses but no longer in the survey
-        const questionIdsSeen = {};
-        questions.forEach(({ id: questionId }) => {
-          questionIdsSeen[questionId] = true;
-        });
-        for (let i = 0; i < surveyResponseAnswers.length; i++) {
-          const extraQuestions = await Promise.all(
-            Object.keys(surveyResponseAnswers[i])
-              .filter(questionId => !questionIdsSeen[questionId])
-              .map(async questionId => {
-                const question = await models.question.findById(questionId);
-                questionIdsSeen[questionId] = true;
-                return question;
-              }),
-          );
-          questions.push(...extraQuestions);
-        }
+      // Exclude 'SubmissionDate' and 'PrimaryEntity' rows from survey response export since these have no answers
+      const questionsForExport = questions.filter(
+        ({ type: questionType }) =>
+          !NON_DATA_ELEMENT_ANSWER_TYPES.includes(questionType) ||
+          questionType === ANSWER_TYPES.INSTRUCTION,
+      );
 
-        // Add the questions info and answers to be exported
-        let exportRow = INFO_ROW_HEADERS.length + 1; // Add one to make up for header row
-        for (let questionIndex = 0; questionIndex < questions.length; questionIndex++) {
-          // Set up the left columns with info about the questions
-          const question = questions[questionIndex];
-          const questionInfo = infoColumnKeys.map(columnKey => question[columnKey]);
-          const questionType = questionInfo[1];
-          // Exclude 'SubmissionDate' and 'PrimaryEntity' rows from survey response export since these have no answers
-          if (
-            NON_DATA_ELEMENT_ANSWER_TYPES.includes(questionType) &&
-            questionType !== ANSWER_TYPES.INSTRUCTION
-          ) {
-            continue;
-          }
-          exportData[exportRow] = questionInfo;
+      // Add the questions info and answers to be exported
+      const preQuestionRowCount = INFO_ROW_HEADERS.length + 1; // Add one to make up for header row
+
+      // Set up the left columns with info about the questions
+      questionsForExport.forEach((question, questionIndex) => {
+        const questionInfo = infoColumnKeys.map(columnKey => question[columnKey]);
+        const exportRow = preQuestionRowCount + questionIndex;
+        exportData[exportRow] = questionInfo;
+      });
+
+      // Add the answers on the right columns to the exportData
+      questionsForExport.forEach((question, questionIndex) => {
+        const exportRow = preQuestionRowCount + questionIndex;
+        surveyResponseAnswers.forEach((answersByQuestionId, surveyResponseIndex) => {
+          const answer = answersByQuestionId[question.id];
+          const exportColumn = infoColumnKeys.length + surveyResponseIndex;
+          exportData[exportRow][exportColumn] = answer || '';
+        });
+      });
```

Comment:
I believe this loop can be consolidated with the one above

Reply:
We separated them intentionally, because they are logically responsible for different things - and it's next to no performance hit to do (x + y) amount of effort n times, vs. x amount of effort n times + y amount of effort n times